src/game/game.cpp

```diff
@@ -1006,11 +1026,10 @@ namespace game
             if(d == focus && third) total *= camera1->o.dist(d->o)/(d != &player1 ? followdist : thirdpersondist);
             int prot = m_protect(gamemode, mutators), millis = d->protect(lastmillis, prot); // protect returns time left
             if(millis > 0) total *= 1.f-(float(millis)/float(prot));
+
             if(d == focus && inzoom())
             {
-                int frame = lastmillis-lastzoom;
-                float pc = frame <= W(d->weapselect, cookzoom) ? (frame)/float(W(d->weapselect, cookzoom)) : 1.f;
-                total *= zooming ? 1.f-pc : pc;
+                total *= 1.f - zoomratio(d);
```

Comment:
I see... kind of non-obvious, but yeah, we're calculating some opacity here... A comment might help.